src/FakeItEasy/Configuration/BuildableCallRule.cs

```diff
@@ -11,7 +11,7 @@ namespace FakeItEasy.Configuration
     internal abstract class BuildableCallRule
         : IFakeObjectCallRule
     {
-        private readonly List<WherePredicate> wherePredicates;
+        private readonly SynchronizedCollection<WherePredicate> wherePredicates;

```

Comment:
Not sure it's necessary here... I think this collection is only modified during configuration of the fake, which is unlikely to be made concurrently from multiple threads